### PR TITLE
Fix IPv6Address.of(byte[]) to correctly deal with the 8th and 16th byte when the first bit is one LOXI-67

### PR DESCRIPTION
**Pull request by ronaldchl**
Reviewer: @rlane @Sovietaced

### CODE REVIEW

**Comment by rlane** [comment]
LGTM

**Comment by bsn-abat** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com//job/loxigen_pull_req/147/
Test PASSed.

**Comment by Sovietaced** [comment]
ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://vss1/abat/2015.05.15.2329-m.loxigen.master/abat.log

File: java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPv6Address.java [hunk 1]
```diff
@@ -174,13 +174,13 @@ public static IPv6Address of(@Nonnull final byte[] address) {
                 (address[0] & 0xFFL) << 56 | (address[1] & 0xFFL) << 48
                         | (address[2] & 0xFFL) << 40 | (address[3] & 0xFFL) << 32
                         | (address[4] & 0xFFL) << 24 | (address[5] & 0xFFL) << 16
-                        | (address[6] & 0xFFL) << 8 | (address[7]);
+                        | (address[6] & 0xFFL) << 8 | (address[7] & 0xFFL);
```
Comment (anshaw) [inline java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPv6Address.java]: Whoops.

File: java_gen/pre-written/src/test/java/org/projectfloodlight/openflow/types/IPv6AddressTest.java [hunk 2]
```diff
@@ -20,7 +20,8 @@
             "::",
             "::1",
             "ffe0::",
-            "1:2:3:4:5:6:7:8"
+            "1:2:3:4:5:6:7:8",
```
Comment (anshaw) [inline java_gen/pre-written/src/test/java/org/projectfloodlight/openflow/types/IPv6AddressTest.java]: And an unfortunate choice of IP address in the unit test!

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://vss1/abat/2015.05.15.2329-m.loxigen.master/abat.log
/cc

File: java_gen/pre-written/src/test/java/org/projectfloodlight/openflow/types/IPv6AddressTest.java (hunk 2 above)
Comment (Sovietaced) [inline java_gen/pre-written/src/test/java/org/projectfloodlight/openflow/types/IPv6AddressTest.java]: indeed.

**Pull request merged**